corehq/apps/userreports/tests/test_choice_provider.py

```diff
@@ -574,6 +582,20 @@ def test_query(self):
                          self.choice_provider.query(ChoiceQueryContext(query='', offset=0)))
         self.assertEqual([], self.choice_provider.query(ChoiceQueryContext(query='D', offset=0)))
 
+    def test_query_no_registry_access(self):
+        role = UserRole.create(
+            domain=self.domain_a,
+            name='Cannot View Registry Data',
+            permissions=Permissions(view_data_registry_contents=False),
+        )
+        self.web_user.set_role(self.domain_a, role.get_qualified_id())
```

Comment:
this will modify the user for this test and all subsequent tests. I would suggest either creating 2 users or else adding a cleanup function for this test to revert the role change.